### PR TITLE
Improvements to documentation

### PR DESCRIPTION
**Pull request by mattmook**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+val anotherOddNumber = oddIterator.next()
+
+// Or, ensure it returns only distinct values
+enum class XYZ { X, Y, Z }
+val enumIterator = fixture.asSequence<XYZ>().distinct().iterator()
+val aDistinctValue = enumIterator.next()
+val anotherDistinctValue = enumIterator.next()
+----
+
+WARNING: As the sequence is infinite, distinct will hang if no more distinct values can be generated.
```
Comment (Rosomack) [inline README.adoc:82]: In a way the sequence will hang in other conditions as well e.g. with an impossible to fulfill `filter` so maybe reword this sentence to something like:

The sequence can hang indefinitely if the applied sequence operators prevent new values to be generated. For example:

* `distinct` hanging if we exhausted the available distinct values. A good practice here is to have a `take(count)` operator limited to the count of distinct values.
* `filter` that can never be fulfilled e.g. `filter { false }`

Or something similar, will leave it to you to work out the wording

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+Everything can be customised, see link:fixture/configuration-options.adoc[configuration options] for more details.
```
Comment (Rosomack) [inline README.adoc:84]: I feel like this is not prominent enough, perhaps make it a section? It will also make it appear in the TOC

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+but to use it with more than just the few basic types built
+into the library requires you to create your own custom generators which
+you then provide.
+
+[source,kotlin]
+._build.gradle.kts_
+----
+// Add for Kotest integration
+testImplementation("com.appmattus.fixture:fixture-kotest:<latest-version>")
+----
```
Comment (Rosomack) [inline README.adoc:99]: This code section would be better placed either at the top or below the next paragraph. Otherwise the 1st paragraph feels quite disconnected. It tells you that there's a problem with Kotest but doesn't tell you how kotlinfixture solves that.

File: README.adoc [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+Additionally, the `javaFakerStrategy` function allows you to override
+faker settings such as `locale`. The available options can be found in
+`link:fixture-javafaker/src/main/kotlin/com/appmattus/kotlinfixture/decorator/fake/javafaker/JavaFakerConfigurationBuilder.kt[JavaFakerConfigurationBuilder]`.
+
+=== Generate strings based on a regex
+
+The module also introduces the ability to generate a random string from
+a Regex, with no need to enable the faker functionality:
```
Comment (Rosomack) [inline README.adoc:155]: This feels like it's not specific to the javafaker integration? Can it be part of the core functionality, or a separate plugin?

File: README.adoc [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+val enumIterator = fixture.asSequence<XYZ>().distinct().iterator()
+val aDistinctValue = enumIterator.next()
+val anotherDistinctValue = enumIterator.next()
+----
+
+WARNING: As the sequence is infinite, distinct will hang if no more distinct values can be generated.
+
+Everything can be customised, see link:fixture/configuration-options.adoc[configuration options] for more details.
+
+== Kotest integration
```
Comment (Rosomack) [inline README.adoc:86]: What do you think about having separate docs for integrations and just have a list with quick summaries here?
e.g.

* link[Kotest integration] - provides fixture-powered generators for property based testing
* link[Javafaker integration] - support for generating values closer matching real data using Javafaker e.g. email strings for email fields

File: fixture-javafaker/README.adoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
+== Java Faker integration
```
Comment (Rosomack) [inline fixture-javafaker/README.adoc:1]: This duplicates all the information from the base readme, ties to my comment about having it as a separate file

File: fixture/comparison.adoc [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+Unlike {link-appmattus}, the other libraries have no understanding of Kotlin's
+https://kotlinlang.org/docs/reference/functions.html#default-arguments[default values]
+which can mean missing an important test scenario.
+
+Only {link-appmattus} has built-in support for abstract classes. Kotlin's https://kotlinlang.org/docs/reference/sealed-classes.html[sealed classes]
+are abstract so not supported, with only {link-marcellogalhardo} supporting
+sealed classes but not abstract classes.
+
+For ease of use, a fixture library needs good support for build-in types to
```
Comment (Rosomack) [inline fixture/comparison.adoc:32]:
```suggestion
For ease of use, a fixture library needs good support for built-in types to
```

File: fixture/comparison.adoc [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+Needless to say all software has bugs, we have tried hard to ensure we create
+core types as expected, and hopefully have enough tests to back that up.
+However, when comparing our library to our competitors we did find
+various unsupported types, not random, or generated invalid types. For example:
+
+- {link-flextrade} does not support `Pair`, `Triple`, `Currency` or `Locale`.
+- `HashMap` is not random in any of the competitors.
+- `SortedSet` and `SortedMap` generate invalid objects in {link-flextrade}.
+- Various `List` types either generate invalid objects or aren't random in {link-easyrandom}.
+- {link-marcellogalhardo} doesn't support primitive arrays such as `ByteArray` and `IntArray`.
```
Comment (Rosomack) [inline fixture/comparison.adoc:47]: Maybe worth having a paragraph mentioning that suggestions are welcome.

**Comment by Rosomack** [comment]
Are there any docs mentioning how to configure the core library (type factories, property factories etc.)

File: README.adoc (hunk 4 above)
Comment (mattmook) [reply README.adoc:155]: was thinking will split it into a separate plugin later on... for today gonna leave it here

File: README.adoc (hunk 3 above)
Comment (mattmook) [reply README.adoc:99]: moving to the top
